File: src/main/java/team4/footwithme/vote/repository/CustomVoteRepository.java
package team4.footwithme.vote.repository;

import team4.footwithme.vote.domain.Vote;
import team4.footwithme.vote.domain.VoteItemLocate;

import java.util.List;
import java.util.Optional;

public interface CustomVoteRepository {

    Optional<Vote> findNotDeletedVoteById(Long id);

    List<Vote> findOpenedVotes();

    Long choiceMemberCountByVoteId(Long voteId);

    Vote findRecentlyVoteByTeamId(Long teamId);

    List<Vote> findAllByTeamId(Long teamId);
}

File: src/main/java/team4/footwithme/vote/repository/CustomVoteRepository.kt
package team4.footwithme.vote.repository

import team4.footwithme.vote.domain.Vote
import java.util.*

interface CustomVoteRepository {
    fun findNotDeletedVoteById(id: Long?): Optional<Vote>

    fun findOpenedVotes(): List<Vote>?

    fun choiceMemberCountByVoteId(voteId: Long?): Long?

    fun findRecentlyVoteByTeamId(teamId: Long?): Vote

    fun findAllByTeamId(teamId: Long?): List<Vote>?
}
